modules/touch/src/touchinteraction.cpp

```diff
@@ -448,41 +440,22 @@ void TouchInteraction::updateStateFromInput(const std::vector<TuioCursor>& list,
         _wasPrevModeDirectTouch = _directTouchMode;
         // evaluates if current frame is in directTouchMode (will be used next frame)
         _directTouchMode =
-            (_currentRadius > _nodeRadiusThreshold && _selected.size() == list.size());
+            (_currentRadius > _nodeRadiusThreshold && _selected.size() == numFingers);
     }
 }
 
-bool TouchInteraction::webContent(const std::vector<TuioCursor>& list) {
-#ifdef OPENSPACE_MODULE_WEBBROWSER_ENABLED
-    glm::ivec2 res = global::windowDelegate.currentWindowSize();
-    glm::dvec2 pos = glm::vec2(
-        list.at(0).getScreenX(res.x),
-        list.at(0).getScreenY(res.y)
-    );
 
-    WebBrowserModule& module = *(global::moduleEngine.module<WebBrowserModule>());
-    return module.eventHandler().hasContentCallback(pos.x, pos.y);
-#else
-    return false;
-#endif
-}
-
-// Activates/Deactivates gui input mode (if active it voids all other interactions)
-bool TouchInteraction::guiMode(const std::vector<TuioCursor>& list) {
+bool TouchInteraction::guiMode(glm::dvec2 screenPosition, size_t numFingers) {
     if (_ignoreGui) {
         return false;
     }
-    glm::ivec2 res = global::windowDelegate.currentWindowSize();
-    glm::dvec2 pos = glm::vec2(
-        list.at(0).getScreenX(res.x),
-        list.at(0).getScreenY(res.y)
-    );
 
     ImGUIModule& module = *(global::moduleEngine.module<ImGUIModule>());
     _guiON = module.gui.isEnabled();
 
-    if (_tap && list.size() == 1 &&
-        std::abs(pos.x) < _guiButton.value().x && std::abs(pos.y) < _guiButton.value().y)
+    if (_tap && numFingers == 1 &&
+        std::abs(screenPosition.x) < _guiButton.value().x
+        && std::abs(screenPosition.y) < _guiButton.value().y)
     {
         // pressed invisible button
```

Comment:
What is this 'invisible button'? Is it referring to the field where the gui is located? Not your code, but does this code comment still make sense?

Reply:
I had no idea, so I left it in.
I guess this is some (test?) code to enable touch interaction with imgui? This PR does not do anything specific with regards to ImGui and touch, but I'll make an issue of it. I'll keep the comment as it might give hints of an invisible button when working on that issue.

Reply:
Added #1039